pkg/proxy/netstat.go

```diff
@@ -0,0 +1,52 @@
+package proxy
+
+import (
+	"fmt"
+	"io/ioutil"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	// procNetTCPFiles is the constant list of /proc/net files to read to get
+	// the same information about open TCP connections as output by netstat.
+	procNetTCPFiles = []string{
+		"/proc/net/tcp",
+		"/proc/net/tcp6",
+	}
+
+	// procNetTCPRegexp matches the first two columns of /proc/net/tcp* files
+	// and submatches on the local port number.
+	procNetTCPRegexp = regexp.MustCompile("^ *[[:digit:]]*: *[[:xdigit:]]*:([[:xdigit:]]*) ")
+)
+
+// readOpenLocalTCPPorts returns the set of TCP ports currently open locally.
+func readOpenLocalTCPPorts() (map[uint16]struct{}, error) {
```

Comment:
An idea - to make this unit-testable, you could add a slice of strings as a parameter to the function. In the actual code base, you could then call this with `procNetTCPFiles` as a parameter. You could then create dummy files for unit testing to ensure that this functions correctly and that we don't break the behavior in the future.

Reply:
Good point. This will also allow using this with UDP, since the format of the `/proc/net/udp*` files is the same. Done.